evennia/contrib/game_systems/achievements/README.md

```diff
@@ -0,0 +1,119 @@
+- **prereqs** (str or list): The *keys* of any achievements which must be completed before this achievement can start tracking progress. An achievement's key is the variable name it's assigned to in your achievement module.
+
+You can add any additional keys to your achievement dicts that you want, and they'll be included with all the rest of the achievement data when using the contrib's functions. This could be useful if you want to add extra metadata to your achievements for your own features.
+
+#### Examples
+
+A simple achievement which you can get just for logging in the first time. This achievement has no prerequisites and it only needs to be fulfilled once to complete, so it doesn't need to define most of the keys.
+```python
+# This achievement has the unique key of "FIRST_LOGIN_ACHIEVE"
+FIRST_LOGIN_ACHIEVE = {
+    "name": "Welcome!", # the searchable, player-friendly display name
+    "desc": "We're glad to have you here.", # the longer description
+    "category": "login", # the type of action this tracks
+    "tracking": "first", # the specific login action
+}
+```
+
+An achievement for killing 10 rats, and another for killing 10 *dire* rats which requires the "kill 10 rats" achievement to be completed first.
+```python
+ACHIEVE_TEN_RATS = {
+    "name": "The Usual",
+    "desc": "Why do all these inns have rat problems?",
+    "category": "defeat",
+    "tracking": "rat",
+    "count": 10,
+}
+
+ACHIEVE_DIRE_RATS = {
+    "name": "Once More, But Bigger",
+    "desc": "Somehow, normal rats just aren't enough any more.",
+    "category": "defeat",
+    "tracking": "dire rat",
+    "count": 10,
+    "prereqs": "ACHIEVE_TEN_RATS",
+}
+```
+
+An achievement for buying 5 each of apples, oranges, and pears.
+```python
+FRUIT_BASKET_ACHIEVEMENT = {
+    "name": "A Fan of Fruit", # note, there is no desc here - that's allowed!
+    "category": "buy",
+    "tracking": ("apple", "orange", "pear"),
+    "count": 5,
+    "tracking_type": "separate",
+}
+```
+
+### `track_achievements`
+
+The primary mechanism for using the achievements system is the `track_achievements` function. In any actions or functions in your game's mechanics which you might want to track in an achievement, add a call to `track_achievements` to update the achievement progress for that individual.
+
+For example, you might have a collection achievement for buying 10 apples, and a general `buy` command players could use. In your `buy` command, after the purchase is completed, you could add the following line:
+
+```python
+    track_achievements(self.caller, "buy", obj.name, count=quantity)
+```
+
+In this case, `obj` is the fruit that was just purchased, and `quantity` is the amount they bought.
+
+The `track_achievements` function does also return a value: an iterable of keys for any achievements which were newly completed by that update. You can ignore this value, or you can use it to e.g. send a message to the player with their latest achievements.
```

Comment:
If I just want to give the player an achievement based on some _external_ condition, how would I go about that? 

Example: Player steps into the dragon's lair for the first time, I independently check this and declare that this means they should get the _Wannabe Dragon Slayer_ achievement.

Reply:
Not sure my question was answered?

Reply:
The very first example provided is an example of an achievement that "just happens".

Reply:
Which one? :thinking: